JavaScript/chapter01/1.5 - OneAway/rroque98_sol.js

```diff
@@ -18,19 +19,32 @@ const isOneAway = (str1, str2) => {
     let errorCount = 0;
     const longestStr = findLongestStr(str1, str2);
     let x = 0;
-    for(let i = 0; i < longestStr; i++) {
-      if (str1[i] !== str2[x]) {
-        errorCount++;
+    for(let i = 0; i < longestStr.length; i++) {
+      let currentChar1 = str1[i];
+      let currentChar2 = str2[x]
+      let nextChar1 = str1[i + 1];
+      let nextChar2 = str2[x + 1];
+      if (str1[i] === str2[x]) {
         x++;
+        continue;
       }
+      errorCount++;
       if (errorCount > 1) {
         return false;
       }
-      x++;
+      if (currentChar1 === nextChar2) {
```

Comment:
having the code handle both cases at the same time is hard to follow. do you think there could be a way in which you only need to deal with the case where both strings are the same length (block L7-L18) and `str1.length < str2.length`?